implementations/micrometer-registry-opentelemetry/src/main/java/io/micrometer/opentelemetry/OpenTelemetryTimer.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright 2020 VMware, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.opentelemetry;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+import io.micrometer.core.instrument.AbstractMeter;
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.Timer;
+import io.micrometer.core.instrument.distribution.Histogram;
+import io.micrometer.core.instrument.distribution.HistogramSnapshot;
+import io.micrometer.core.instrument.distribution.NoopHistogram;
+import io.opentelemetry.common.Labels;
+import io.opentelemetry.metrics.DoubleValueRecorder;
+
+public class OpenTelemetryTimer extends AbstractMeter implements Timer {
+    final DoubleValueRecorder recorder;
```

Comment:
Should there be one of these for "sum", "count", "max", etc rather than just one for what appears to be effectively just "sum"?

Reply:
This is a question I asked yesterday. The Java API doesn't carry over the spec descriptions in the Javadoc fully. According to the spec: 

> The default aggregation for ValueRecorder computes the minimum and maximum values, the sum of event values, and the count of events, allowing the rate, the mean, and range of input values to be monitored.

https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/metrics/api.md#valuerecorder

Reply:
There is still ongoing discussion about what the default aggregation should be for ValueRecorders (it's MMSC today). And, there is allegedly a views API coming which would allow any sort of customization of the aggregation for a given instrument type. I did a prototype of a proto-views API in this (very out of date, stale) PR:  https://github.com/open-telemetry/opentelemetry-java/pull/1412

Reply:
Point stands.. I only need the one instrument in this case. ;)